### ISSUE TITLE
index.ts and Circular dependency

### ISSUE THREAD

**Issue by eosimosu** [jhipster/generator-jhipster#8170]
##### **Overview of the feature request**

Barrels aka index.ts leads to maintenance hell and circular references that are sometimes extremely difficult to debug. 

##### **Motivation for or Use Case**

index.ts files were very useful prior to introduction of NgModules as they helped manage countless imports, but now they are obsolete and just causes a lot of headaches due to circular dependency that are sometimes extremely difficult to debug. 

More about this issue here: [https://github.com/angular/angular-cli/issues/7369](https://github.com/angular/angular-cli/issues/7369)

I suggest they should be removed in favor of direct import.

##### **Related issues or PR**

- [ x] Checking this box is mandatory (this is just to show you read everything)

**Comment by deepu105** [comment]
You are right, I started adding barrels as per the Angular style guide recomendation at that time and soon realized that it was annoying. But I dont have the time to do the refactoring so if you is willing to do the refactoring you are most welcome.

**Comment by eosimosu** [comment]
Great! I'd do the refactoring

**Comment by jdubois** [comment]
Awesome! Thanks @eosimosu

**Comment by vishal423** [comment]
I think barrels are removed from angular style guide in favor of NgModule, however, I still find them quite useful for cleaner imports. If you follow discipline, you will never run into circular dependency issue.

**Comment by eosimosu** [comment]
@jdubois I started to work on this but haven't had time to finish it.
I'll re-open this issue when it's done.

**Comment by jdubois** [comment]
Oh, awesome -> yes don't hesitate!!!! No pressure on you, when you have time to do it, that's perfect.

**Comment by kaidohallik** [comment]
I opened PR #10185 which removes circular dependencies. I would like to extend this PR to completely remove barrels. Is this ok?

**Comment by vishal423** [comment]
@kaidohallik, please go ahead. Should I re-open this ticket and assign to you?

**Comment by kaidohallik** [comment]
@vishal423, for me it's not necessary to re-open this ticket. I added commit to PR which removes barrels, so hopefully work is done. Let's see if CI tests will pass.

**Comment by deepu105** [comment]
Yes lets remove barrels. I added them since it was official recomendation
from the Angular style guide, but they have removed that recomendation
since I guess.

Thanks & Regards,
Deepu
